src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.cs

```diff
@@ -127,7 +127,11 @@ public static NamingStylePreferences ParseDictionary(IReadOnlyDictionary<string,
 
         private static Dictionary<string, string> TrimDictionary(IReadOnlyDictionary<string, string> allRawConventions)
         {
-            var trimmedDictionary = new Dictionary<string, string>(allRawConventions.Count);
+            // At this point, keys have been lowercased. So, for example, to make a naming style
+            // called "Pascal_Case_style" match up correctly with the key
+            // "dotnet_naming_style.pascal_case_style.capitalization", we have to ignore casing for
+            // lookups in this dictionary.
+            var trimmedDictionary = new Dictionary<string, string>(allRawConventions.Count, StringComparer.OrdinalIgnoreCase);
```

Comment:
This should use the KeyComparer property so we're definitely using the same comparer as before.